ApplicationCode/ProjectDataModel/RimElementVectorResult.cpp

```diff
@@ -181,30 +262,157 @@ const cvf::Color3f& RimElementVectorResult::getUniformVectorColor() const
+                directions++;
+                unitVectors.push_back( cvf::Vec3d( 0, 1, 0 ) );
+            }
+            cleanedResVarAddresses.push_back( resVarAddresses.at( 1 + fluidIndex ) );
+        }
+        if ( showVectorK() )
+        {
+            if ( fluidIndex == 0 )
+            {
+                directions++;
+                unitVectors.push_back( cvf::Vec3d( 0, 0, 1 ) );
+            }
+            cleanedResVarAddresses.push_back( resVarAddresses.at( 2 + fluidIndex ) );
+        }
+    }
+    resVarAddresses = cleanedResVarAddresses;
 
-    if ( m_legendConfig->rangeMode() == RimRegularLegendConfig::RangeModeType::AUTOMATIC_ALLTIMESTEPS )
+    if ( directions > 0 )
     {
-        resultsData->minMaxCellScalarValues( resVarAddr, min, max );
+        std::vector<double> directionsMax;
+        directionsMax.resize( directions, 0.0 );
+        std::vector<double> directionsMin;
+        directionsMin.resize( directions, 0.0 );
+
+        for ( size_t index = 0; index < resVarAddresses.size(); index += directions )
+        {
+            cvf::Vec3d aggregatedVectorMax;
+            cvf::Vec3d aggregatedVectorMin;
+            for ( size_t dir = 0; dir < directions; dir += 1 )
+            {
+                double localMin = cvf::UNDEFINED_DOUBLE;
+                double localMax = cvf::UNDEFINED_DOUBLE;
+
+                RigEclipseResultAddress resVarAddr = resVarAddresses.at( index + dir );
+                if ( !resVarAddr.isValid() ) return;
+
+                RimEclipseView*         eclipseView = dynamic_cast<RimEclipseView*>( view );
+                RigCaseCellResultsData* resultsData =
+                    eclipseView->eclipseCase()->eclipseCaseData()->results( RiaDefines::PorosityModelType::MATRIX_MODEL );
+
+                resultsData->ensureKnownResultLoaded( resVarAddr );
+
+                if ( m_legendConfig->rangeMode() == RimRegularLegendConfig::RangeModeType::AUTOMATIC_ALLTIMESTEPS )
+                {
+                    resultsData->minMaxCellScalarValues( resVarAddr, localMin, localMax );
+                }
+                else if ( m_legendConfig->rangeMode() == RimRegularLegendConfig::RangeModeType::AUTOMATIC_CURRENT_TIMESTEP )
+                {
+                    resultsData->minMaxCellScalarValues( resVarAddr, currentTimeStep, localMin, localMax );
+                }
+                if ( vectorView() == RimElementVectorResult::VectorView::AGGREGATED )
+                {
+                    aggregatedVectorMax += unitVectors.at( dir ) * localMax;
+                    aggregatedVectorMin += unitVectors.at( dir ) * localMin;
+                }
+                else
+                {
+                    directionsMax[dir] += localMax;
+                    directionsMin[dir] += localMin;
+                }
+            }
+            if ( vectorView() == RimElementVectorResult::VectorView::AGGREGATED )
+            {
+                directionsMax[0] += aggregatedVectorMax.length();
```

Comment:
Not sure I understand why only directionX is scaled?

Reply:
It's not only X being scaled. If you look further down in the code, only the first value of directionsMin/directionsMax is used for aggregated vectors. Aggregated vectors do not have single i, j, k values, but are a combination of all directions. Hence, we do only need to have a look at one value per fluid.